### PR TITLE
feat(core): add CI generation to create-nx-workspace

### PR DESCRIPTION
**Pull request by meeroslav**
This PR adds the functionality of generating CI workflow:

* Add CI workflow generation to create-nx-workspace
* Add CI workflow generator to @nrwl/workspace
* Support GitHub, Azure, and CircleCI
* Update monorepo guides

## Related Issue(s)

Fixes #

### CODE REVIEW

File: docs/generated/packages/workspace.json [hunk 1]
```diff
@@ -654,6 +654,48 @@
       "aliases": [],
       "hidden": false,
       "path": "/packages/workspace/src/generators/npm-package/schema.json"
+    },
+    {
+      "name": "ci-workflow",
+      "factory": "./src/generators/ci-workflow/ci-workflow#ciWorkflowGenerator",
+      "schema": {
+        "$schema": "http://json-schema.org/schema",
+        "$id": "NxWorkspaceNpmPackage",
```
Comment (AgentEnder) [inline docs/generated/packages/workspace.json:663]: This seems wrong?

File: packages/create-nx-workspace/bin/create-nx-workspace.ts [hunk 2]
```diff
@@ -605,6 +624,75 @@ async function determineStyle(
...
+      .prompt([
+        {
+          name: 'CI',
+          message: `Autogenerate CI workflow file (multi-select)?`,
+          type: 'multiselect',
+          choices: [
+            { message: 'GitHub Actions', name: 'github' },
+            { message: 'Circle CI', name: 'circleci' },
+            { message: 'Azure DevOps', name: 'azure' },
+          ],
```
Comment (AgentEnder) [inline packages/create-nx-workspace/bin/create-nx-workspace.ts:688]: Is there a way to select none?

**Review by AgentEnder — CHANGES_REQUESTED** [review]

File: packages/create-nx-workspace/bin/create-nx-workspace.ts (hunk 2 above)
Comment (meeroslav) [reply packages/create-nx-workspace/bin/create-nx-workspace.ts:688]: Yes, it's a multi-select so you can select 0..3

File: docs/generated/packages/workspace.json (hunk 1 above)
Comment (meeroslav) [reply docs/generated/packages/workspace.json:663]: Good catch, I will fix it

File: packages/create-nx-workspace/bin/create-nx-workspace.ts (hunk 2 above)
Comment (meeroslav) [reply packages/create-nx-workspace/bin/create-nx-workspace.ts:688]: That is also the default value

**Review by AgentEnder — APPROVED** [review]

**Pull request merged**